docs/APIReference-Route.md

```diff
@@ -70,6 +76,57 @@ class ProfileRoute extends Relay.Route {
 }
 ```
 
+### prepareParams (static property)
+
+```
+static prepareParams: ?(prevParams: {[prevParam: string]: mixed}) => {[param: string]: mixed};
+```
+
+Routes can declare additional parameters, or conversion for parameters on initialisation. These can also be used as query parameters in your nested containers.
+
+#### Example
+
+```
+class ProfileRoute extends Relay.Route {
+  static queries = {
+    viewer: () => Relay.QL`query { viewer }`
+  };
+  static prepareParams = (prevParams) => {
+    return {
+      // Ensure any params that ProfileRoute gets initialised with will still be passed down.
+      ...prevParams,
+      //  Override the `limit` variable in the top level container.
+      limit: 10
+    }
+  }
+  // ...
+}
+
+const Container = Relay.createContainer(({viewer}) => (
+  <View>
+    {viewer.edges.map((edge) => <div>edge.node.name</div>)}
+  </View>
+  ), {
+  initialVariables: {
+    limit: null
+  },
+  fragments: {
+    viewer: () => Relay.QL`
+      fragment on Viewer {
+        friends(first: $limit) {
+          edges {
+            node {
+              name
+            }
+          }
+        }
+      }
+    `
+  }
+});
+```
+In this example `prepareParams` overrides any `limit` variables the route gets initialised with. Given `ProfileRoute` and `Container` are both being passed into `Relay.Renderer`, that limit variable will automatically be passed down to `Container`.
+
```

Comment:
@Globegitter on second reading, this example conflates two things: `prepareParams` and passing route vars to the root container fragment.

Can you separate these? Note that `prepareParams` can be used for things like setting default values or converting from strings (e.g. in a URL) to integers. I'd recommend keeping this PR focused on `prepareParams` only, and maybe add a follow-up for the route -> root container variable example.